dolphinscheduler-api/src/main/resources/application.yaml

```
@@ -143,6 +143,8 @@ security:
       password: password
       user.identity.attribute: uid
       user.email.attribute: mail
+      # action when ldap user is not exist (supported types: CREATE,DENY)
+      user.not.exist.action: CREATE
```

1. Please use YAML hierarchy to re-organize the configs (including `user.email.attribute`, `user.identity.attribute`, `user.admin`), and you should rename the config a little bit (not using `.` to split as in YAML that will become a new section), an example below

```suggestion
      user:
        admin: ""
        email-attribute: ""
        identity-attribute: ""
        absent-policy: CREATE
```

2. When adding a new config, make sure its default value is the same as the one before you add this config. Previously the policy is `DENY` so you should  set the default value of the new config as `DENY`, so that users don't face different behaviors after upgrading.

---

Thanks for your reminder. I have reorganized the YAML configs. And the default policy is `create`, when a user does not exists will auto create a user when using LDAP log-in.